### PR TITLE
Feature/favorites screen

### PR DESCRIPTION
**Pull request by vasileGrad**

### CODE REVIEW

File: app/src/main/java/com/example/hungrywolfs/model/DetailsViewModel.kt [hunk 1]
```diff
@@ -15,17 +16,28 @@ class DetailsViewModel : ViewModel() {
...
+        _favoriteMeals.value = Hawk.get<MutableList<MealDetailsInfo>>("favorites")
+    }
+
     fun getMealWithDetails(idMeal: String) {
         viewModelScope.launch {
             try {
                 _mealDetails.value =
                     MealApi.retrofitService.getMealDetails(idMeal).meals.getOrNull(0)
+                if (_favoriteMeals.value?.contains(_mealDetails.value) == true) {
+                    isFavorite.value = true
```
Comment (mgabbi) [inline app/src/main/java/com/example/hungrywolfs/model/DetailsViewModel.kt]: You can reduce all of this to `isFavorite.value = _favoriteMeals.value?.contains(_mealDetails.value)`
Technically you assign true when other value is true, so the check is not necessary

File: app/src/main/java/com/example/hungrywolfs/model/FavoritesViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import com.example.hungrywolfs.network.MealDetailsInfo
+import com.orhanobut.hawk.Hawk
+
+class FavoritesViewModel : ViewModel() {
+
+    private val _favoriteMeals = MutableLiveData<MutableList<MealDetailsInfo>>()
+    val favoriteMeals: LiveData<MutableList<MealDetailsInfo>> = _favoriteMeals
+
+    init {
+        _favoriteMeals.value = Hawk.get<MutableList<MealDetailsInfo>>("favorites")
```
Comment (mgabbi) [inline app/src/main/java/com/example/hungrywolfs/model/FavoritesViewModel.kt]: You can call getFavorites() to remove duplicate code here

**Review by mgabbi — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/hungrywolfs/model/DetailsViewModel.kt [hunk 3]
```diff
@@ -15,17 +16,28 @@ class DetailsViewModel : ViewModel() {
     private var _mealDetails = MutableLiveData<MealDetailsInfo>()
     val mealDetails: LiveData<MealDetailsInfo> = _mealDetails
 
+    private var _favoriteMeals = MutableLiveData<MutableList<MealDetailsInfo>>()
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/model/DetailsViewModel.kt]: No need to use a livedata here, it could be just a list.

**Review by cristiwd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/hungrywolfs/Constants.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.hungrywolfs
+
+class Constants {
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/Constants.kt]: Why not object?

Comment (vasileGrad) [reply app/src/main/java/com/example/hungrywolfs/Constants.kt]: I removed the class Constants.

Comment (vasileGrad) [reply app/src/main/java/com/example/hungrywolfs/Constants.kt]: Using the first approach with the companion object inside the class will generate getter and setter instance methods for the fields to be accessible. Calling instance methods is technically more expensive than calling static methods. For this reason, the recommended practice is to define the constants in the **object**.

**Review by cristiwd — APPROVED** [review]

**Review by mgabbi — APPROVED** [review]

**Pull request merged**